src/shapes/Group.ts

```diff
@@ -644,21 +650,14 @@ export class Group
       enlivenObjects<FabricObject>(objects),
       enlivenObjectEnlivables(options),
     ]).then(([objects, hydratedOptions]) => {
-      const restoredGroup = new this(
-        objects,
-        {
-          ...options,
-          ...hydratedOptions,
-        },
-        true
-      );
-      if (!options.strategy) {
-        // restore the old save width/height killed by the
-        // default layour manager
-        restoredGroup.width = options.width;
-        restoredGroup.height = options.height;
-      }
-      return restoredGroup;
+      const group = new this(objects, {
+        ...options,
+        ...hydratedOptions,
+        layoutManager: new RespectfulLayoutManager(),
+      });
+      group.layoutManager = new LayoutManager();
+      group.setCoords();
```

Comment:
do we need this?

Reply:
yes because is the layoutManager that is doing setCoords for the group initialization.
It doesn't have to, but is doing it, it changes width and height or top and left, so it make sense it also calls setCoords.
If the the layoutManager is a no-op then a manual call to setCoords is necessary

Reply:
The layout manager did not call `setCoords` in the intialization layout.
  
<img width="1136" alt="Screenshot 2023-12-29 at 8 02 36" src="https://github.com/fabricjs/fabric.js/assets/34343793/726fd75a-1c5b-4464-b84a-52e47aab5825">